subsys/power_seq/x86_non_dsx.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* On power-on start boot up sequence */
+static enum power_states_ndsx power_state = SYS_POWER_STATE_G3S5;
+
+/* Implement in board file */
+static void enable_power_rails(bool enable)
+{
+	if (enable) {
+		gpio_set_level(AP_5P0V_A_RAIL, 1);
```

Comment:
We should probably define these gpios in a power-sequence device in the device tree. This code would use that device information and expect certain gpio names to be populated in the device tree, like en-ap-5p0v-a. Then this cod would look that up and use the gpio_* Zephyr API (e.g `gpio_pin_get_raw` `gpio_port_set_bits_raw` and `gpio_port_clear_bits_raw`) to interact with the gpios.

Reply:
> How do we call / configure this code? Is there an example?

Added sample code

Reply:
> We should probably define these gpios in a power-sequence device in the device tree. This code would use that device information and expect certain gpio names to be populated in the device tree, like en-ap-5p0v-a. Then this cod would look that up and use the gpio_* Zephyr API (e.g `gpio_pin_get_raw` `gpio_port_set_bits_raw` and `gpio_port_clear_bits_raw`) to interact with the gpios.

ACK, added device tree now, I will work on adding follow-on CLs. Try to learn this new interface